From bb9de6b7eab6b722af36dd96501e72d76a6ad30e Mon Sep 17 00:00:00 2001
From: Shaun Tancheff <shaun.tancheff@seagate.com>
Date: Fri, 6 May 2016 20:01:48 -0500
Subject: [PATCH 64/65] Fix intramodule dependencies

blk-zoned needs scsi-zbc
dm-zoned needs blk-zoned (and scsi-zbc)

dm-zdm needs crc16.

Signed-off-by: Shaun Tancheff <shaun.tancheff@seagate.com>
---
 block/Kconfig      |  1 +
 drivers/md/Kconfig | 18 +++++++++++-------
 2 files changed, 12 insertions(+), 7 deletions(-)

diff --git a/block/Kconfig b/block/Kconfig
index 8162312..583eb6e 100644
--- a/block/Kconfig
+++ b/block/Kconfig
@@ -116,6 +116,7 @@ config BLK_DEV_THROTTLING
 config BLK_DEV_ZONED
 	bool "Zoned block device support"
 	default n
+	select SCSI_ZBC
 	---help---
 	Block layer zoned block device support. This option enables
 	support for zoned block (ZAC/ZBC) devices.
diff --git a/drivers/md/Kconfig b/drivers/md/Kconfig
index b2034cd..ac1d31f 100644
--- a/drivers/md/Kconfig
+++ b/drivers/md/Kconfig
@@ -339,6 +339,9 @@ config DM_ERA
 config DM_ZONED
 	tristate "Zoned cache write target support (EXPERIMENTAL)"
 	depends on BLK_DEV_DM
+	default n
+	select BLK_DEV_ZONED
+	select SCSI_ZBC
 	---help---
 	 This device-mapper target implements a caching layer for
 	 zoned block devices (ZBC).
@@ -349,19 +352,20 @@ config DM_ZONED
 	 If unsure, say N.
 
 config DM_ZDM
-       tristate "ZDM: Zoned based device target (EXPERIMENTAL)"
-       depends on BLK_DEV_DM
-       default n
-       select LIBCRC32C
-       ---help---
-         This device-mapper target implements a translation layer
+	tristate "ZDM: Zoned based device target (EXPERIMENTAL)"
+	depends on BLK_DEV_DM
+	default n
+	select LIBCRC32C
+	select CRC16
+	---help---
+	 This device-mapper target implements a translation layer
 	 for zoned block devices (ZBC/ZAC)
 	 ZDM performs write serialization and copy-on-write data to
 	 ensure forward writing within zones. ZDM also includes a
 	 garbage collection model to reclaim stale blocks and remap
 	 in use blocks.
 
-         Use zdmadm to create, repair and/or restore ZDM instances.
+	 Use zdmadm to create, repair and/or restore ZDM instances.
 
 	 To compile this code as a module, choose M here: the module will
 	 be called dm-zoned.
-- 
2.8.1

